src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -710,6 +710,7 @@ def test_list_template_spec(self, resource_group, resource_group_one, resource_g
         self.cmd('ts delete --template-spec {template_spec_id_rg1} --yes')
 
     @ResourceGroupPreparer(name_prefix='cli_test_template_specs', location='westus')
+    @live_only()
```

Comment:
Because the path separator of the operating system for recording yaml files may be different from that of the current operating system, the test should be changed to `@live_only()`

Reply:
I think we should not add live_only in such scenario. If there is only path separator issue, we should make the code compatible for different OS, not using hard coded path separator.

Reply:
```
self.check('artifacts[0].path', 'artifacts' + os.sep + 'createResourceGroup.json')
```

In fact the scenario why I add `live_only()` is like this:
1. When the test is executed locally on Windows system, the path separator is `\`, and the value recorded in the yaml file is "artifacts\createResourceGroup.json"
2. However, CI runs the test on a Linux system, and the path separator is `/`, so the expected match value is "artifacts/createResourceGroup.json". Therefore, there will be a error that the check does not match.

Because the path separator will be recorded into the yaml file, so the test of record mode in different operating systems will have the problem of mismatch.

Reply:
Instead of using live_only, can we use os-independent check?@Zhou.Xing@microsoft.com 

---
In reply to: [522156993](https://github.com/Azure/azure-cli/pull/15782#discussion_r522156993) [](ancestors = 522156993)

Reply:
using check with `contains` not `equal` also would be fine.

Reply:
OK, I use the regular expression validation instead